File: leetcode-codes/0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
	int maxProfit(vector<int>& prices) {
		int maxsell = 0;
		int minstock = prices[0];
		for (int i = 1; i < prices.size(); i++) {
			minstock=min(prices[i] , minstock);
			maxsell = max(maxsell, prices[i] - minstock);
		}
		return maxsell;
	}
};

File: leetcode-codes/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution(object):
    def maxProfit(self, prices):
        _min = prices[0]
        res = 0
        for i in range(1,len(prices)):
            res = max(prices[i]-_min,res)
            _min =min(_min,prices[i])
        return res
        
